[Feature] Database code refactor (#13589)
jellyfin/jellyfin: src/Jellyfin.Database/readme.md

# How to run EFCore migrations

> This shall provide context on how to work with entity frameworks multi provider migration feature.

Reviewer: ```suggestion
This shall provide context on how to work with Entity Framework's multi provider migration feature.
```

> Jellyfin will in the future support multiple database providers, namely SqLite as its default and the experimental postgresSQL.

Reviewer: ```suggestion
Jellyfin will support multiple database providers in the future, namely SqLite as its default and the experimental postgresSQL.
```

Not sure if we should write this here since postgres won't be in 10.11

Author: I mean ... future ... could also write it as "soon ™"

> Jellyfin will support multiple database providers in the future, namely SqLite as its default and the experimental postgresSQL.

Reviewer 2: ```suggestion
Jellyfin will support multiple database providers in the future, namely SQLite as its default and the experimental PostgreSQL.
```

Each provider has its own set of migrations, as they contain provider specific instructions to migrate the specific changes to their respective systems.

When creating a new migration, you always have to create migrations for all providers. This is supported via the following syntax:

```cmd
dotnet ef migrations add MIGRATION_NAME --project "PATH_TO_PROJECT" -- --provider PROVIDER_KEY
```

> with sqlite currently beeing supported and, you need to run the Entity Framework tool with the correct project to tell EFCore where to store the migrations and the correct provider key to tell jellyfin to load that provider.

Reviewer: ```suggestion
with sqlite currently being the only supported provider, you need to run the Entity Framework tool with the correct project to tell EFCore where to store the migrations and the correct provider key to tell jellyfin to load that provider.
```

> with sqlite currently being the only supported provider, you need to run the Entity Framework tool with the correct project to tell EFCore where to store the migrations and the correct provider key to tell jellyfin to load that provider.

Reviewer: ```suggestion
With SQLite currently being the only supported provider, you need to run the Entity Framework tool with the correct project to tell EFCore where to store the migrations and the correct provider key to tell Jellyfin to load that provider.
```

> with sqlite currently being the only supported provider, you need to run the Entity Framework tool with the correct project to tell EFCore where to store the migrations and the correct provider key to tell jellyfin to load that provider.

Reviewer 2: ```suggestion
with SQLite currently being the only supported provider, you need to run the Entity Framework tool with the correct project to tell EFCore where to store the migrations and the correct provider key to tell Jellyfin to load that provider.
```

> The example is made from the root folder of the project e.g for codespaces `/workspaces/jellyfin`

Reviewer: Incorrect since it was moved to `src`

Author: well no, that example is just for where you are by default on a console when running it in codespaces. The path is correct.

```cmd
dotnet ef migrations add {MIGRATION_NAME} --project "Jellyfin.Database/Jellyfin.Database.Providers.SqLite" -- --migration-provider Jellyfin-SQLite
```

If you get the error: `Run "dotnet tool restore" to make the "dotnet-ef" command available.` Run `dotnet restore`.

> in the event that you get the error: `System.UnauthorizedAccessException: Access to the path '/Jellyfin.Database' is denied.` you have to restore as sudo and then run `ef migrations` as sudo too.

Reviewer: Wrong path too
